fast64_internal/oot/cutscene/classes.py

```diff
@@ -0,0 +1,597 @@
+import bpy
+
+from dataclasses import dataclass, field
+from bpy.types import Object
+from typing import Optional
+from ..oot_constants import ootData
+from .motion.utility import getBlenderPosition, getBlenderRotation, getRotation, getInteger
+
+
+# NOTE: ``paramNumber`` is the expected number of parameters inside the parsed commands,
+# this account for the unused parameters. Every classes are based on the commands arguments from ``z64cutscene_commands.h``
+
+
+@dataclass
+class CutsceneCmdBase:
+    """This class contains common Cutscene data"""
+
+    params: list[str]
+
+    startFrame: Optional[int] = None
+    endFrame: Optional[int] = None
+
+    def getEnumValue(self, enumKey: str, index: int, isSeqLegacy: bool = False):
+        enum = ootData.enumData.enumByKey[enumKey]
+        item = enum.itemById.get(self.params[index])
+        if item is None:
+            setting = getInteger(self.params[index])
+            if isSeqLegacy:
+                setting -= 1
+            item = enum.itemByIndex.get(setting)
+        return item.key if item is not None else self.params[index]
+
+
+@dataclass
+class CutsceneCmdCamPoint(CutsceneCmdBase):
+    """This class contains a single Camera Point command data"""
+
+    continueFlag: Optional[str] = None
+    camRoll: Optional[int] = None
+    frame: Optional[int] = None
+    viewAngle: Optional[float] = None
+    pos: tuple[int, int, int] = field(default_factory=tuple)
+    paramNumber: int = 8
+
+    def __post_init__(self):
+        if self.params is not None:
+            self.continueFlag = self.params[0]
+            self.camRoll = getInteger(self.params[1])
+            self.frame = getInteger(self.params[2])
+            self.viewAngle = float(self.params[3][:-1])
+            self.pos = [getInteger(self.params[4]), getInteger(self.params[5]), getInteger(self.params[6])]
```

Comment:
Sorry, one more thing! You changed the types to tuples, but what you're actually storing in it is a list. I don't know what code accesses this, but if it needs to be a list rather than a tuple then the type should be `list[int]`. Otherwise change the brackets on line 51 to parentheses. (Same for the other places in this file)